develop/advanced_plugin_config.rst

```diff
@@ -1,174 +1,142 @@
-Advanced Plugin Config
-**********************
-This tutorial covers some of the more advanced plugin config features available
-to Supybot plugin authors.
-
-What's This Tutorial For?
-=========================
-  Brief overview of what this tutorial covers and the target audience.
-
-Want to know the crazy advanced features available to you, the Supybot plugin
-author? Well, this is the tutorial for you. This article assumes you've read
-the Supybot plugin author tutorial since all the basics of plugin config are
-handled there first.
-
-In this tutorial we'll cover:
-
-* Creating config variable groups and config variables underneath those
-  groups.
-* The built-in config variable types ("registry types") for use with config
-  variables
-* Creating custom registry types to handle config variable values more
-  effectively
-* Using the configure function for interactive configuration in supybot-wizard
-
-Using Config Groups
-===================
-  A brief overview of how to use config groups to organize config variables
+***********************************
+Plugin Configuration for Developers
+***********************************
+
+This page describes how to use Limnoria's config system when developing plugins.
+For the user guide, see the :ref:`Configuration page <configuration-guide>` instead.
+
+.. contents::
 
-Supybot's Hierarchical Configuration
-------------------------------------
+Adding Plugin Configuration (config.py)
+=======================================
 
-Supybot's configuration is inherently hierarchical, as you've probably already
-figured out in your use of the bot. Naturally, it makes sense to allow plugin
-authors to create their own hierarchies to organize their configuration
-variables for plugins that have a lot of plugin options. If you've taken a look
-at the plugins that Supybot comes with, you've probably noticed that several of
-them take advantage of this. In this section of this tutorial we'll go over how
-to make your own config hierarchy for your plugin.
+Limnoria's configuration is hierarchical. In most cases, plugins will register
+its configuration under a group reflecting its name: ``supybot.plugins.<pluginname>``.
+The default template provided by :command:`supybot-plugin-doc` will already set
+this up, e.g.::
 
-Here's the brilliant part about Supybot config values which makes hierarchical
-structuring all that much easier - values are groups. That is, any config value
-you may already defined in your plugins can already be treated as a group, you
-simply need to know how to add items to that group.
+    WorldDom = conf.registerPlugin('WorldDom')
 
-Now, if you want to just create a group that doesn't have an inherent value you
-can do that as well, but you'd be surprised at how rarely you have to do that.
-In fact if you look at most of the plugins that Supybot comes with, you'll only
-find that we do this in a handful of spots yet we use the "values as groups"
-feature quite a bit.
```

Comment:
we should keep at least a paragraph about variables being hierarchical

Reply:
Hmm, I think that is already described in depth in the user facing guide. How much more detail do we need to add here?

Reply:
there's a French saying: "pedagogy is repetition".

Developers may have missed it, or forgot about it, since they read the user guide. There is no harm in a short reminder.